src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/SymbolIsBannedInAnalyzersAnalyzer.cs

```diff
@@ -0,0 +1,121 @@
+            void analyzeAnalyzersAndGeneratorsIfPropertyNotSpecified(CompilationStartAnalysisContext context)
+            {
+                var propertyValue = context.Options.GetMSBuildPropertyValue(MSBuildPropertyOptionNames.EnforceExtendedAnalyzerRules, context.Compilation);
+                if (propertyValue is not null)
+                {
+                    return;
+                }
+
+                var provider = WellKnownTypeProvider.GetOrCreate(context.Compilation);
+                var diagnosticAnalyzerAttributeType = provider.GetOrCreateTypeByMetadataName(WellKnownTypeNames.MicrosoftCodeAnalysisDiagnosticsDiagnosticAnalyzerAttribute);
+                var generatorAttributeType = provider.GetOrCreateTypeByMetadataName(WellKnownTypeNames.MicrosoftCodeAnalysisGeneratorAttribute);
+                context.RegisterSymbolAction(analyzePossibleAnalyzerOrGenerator, SymbolKind.NamedType);
+
+                void analyzePossibleAnalyzerOrGenerator(SymbolAnalysisContext symbolAnalysisContext)
+                {
+                    var symbol = symbolAnalysisContext.Symbol;
+
+                    var attributes = symbol.GetAttributes();
+                    if (attributes.Any(shouldReportNotSpecifiedEnforceAnalyzerBannedApisSetting))
+                    {
+                        symbolAnalysisContext.ReportDiagnostic(symbol.Locations.CreateDiagnostic(SymbolIsBannedInAnalyzersAnalyzer.NoSettingSpecifiedSymbolIsBannedRule, symbol));
+                    }
+
+                    bool shouldReportNotSpecifiedEnforceAnalyzerBannedApisSetting(AttributeData attributeData)
+                    {
+                        return attributeData.AttributeClass.Equals(diagnosticAnalyzerAttributeType, SymbolEqualityComparer.Default)
+                            || attributeData.AttributeClass.Equals(generatorAttributeType, SymbolEqualityComparer.Default);
+                    }
+                }
+            }
+        }
+
+#pragma warning disable RS1012 // 'compilationContext' does not register any analyzer actions. Consider moving actions registered in 'Initialize' that depend on this start action to 'compilationContext'.
+        protected sealed override Dictionary<ISymbol, BanFileEntry>? ReadBannedApis(CompilationStartAnalysisContext compilationContext)
+        {
+            var propertyValue = compilationContext.Options.GetMSBuildPropertyValue(MSBuildPropertyOptionNames.EnforceExtendedAnalyzerRules, compilationContext.Compilation);
+            if (propertyValue != "true")
+            {
+                return null;
+            }
+
+            const string fileName = "Microsoft.CodeAnalysis.Analyzers.AnalyzerBannedSymbols.txt";
+            var stream = typeof(SymbolIsBannedInAnalyzersAnalyzer<>).Assembly.GetManifestResourceStream(fileName);
+            var source = SourceText.From(stream);
+            var result = new Dictionary<ISymbol, BanFileEntry>();
```

Comment:
Shouldn't we specify a symbol equality comparer here? Believe we have an anlyzer that tries to enforce that.

Reply:
We use this same method of making the dictionary in [main](https://github.com/dotnet/roslyn-analyzers/blob/main/src/Microsoft.CodeAnalysis.BannedApiAnalyzers/Core/SymbolIsBannedAnalyzer.cs#L207), but maybe we should start using [SymbolEqualityComparer.Default](https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.symbolequalitycomparer.default?view=roslyn-dotnet-4.2.0) here.

Reply:
```suggestion
            var result = new Dictionary<ISymbol, BanFileEntry>(SymbolEqualityComparer.Default);
```